src/escalus_connection.erl

```diff
@@ -177,25 +183,44 @@ get_stanza(Conn, Name) ->
     get_stanza(Conn, Name, ?TIMEOUT).
 
 -spec get_stanza(client(), any(), timeout()) -> exml_stream:element().
-get_stanza(#client{rcv_pid = Pid, jid = Jid}, Name, Timeout) ->
+get_stanza(Client, Name, Timeout) ->
+    case get_stanza_safe(Client, Timeout) of
+        {error, timeout} ->
+            throw({timeout, Name});
+        {Stanza, _} ->
+            Stanza
+    end.
+
+-spec get_stanza_with_metadata(client(), any(), timeout()) -> {exml_stream:element(), map()}.
```

Comment:
Would it be possible to spec metadata map more precisely? I see only `recv_timeout` key, so I guess we could add a spec for that. Is it possible to somehow add own metadata keys?

Reply:
The problem with map spec is that it's supported from Erlang/OTP 19 while escalus is officially tested with Erlang/OTP 18 to 20. I think more precise type could be added when we ditch support for 18.

Reply:
Currently it's not possible to add your own metadata. How would you see it?

Reply:
@michalwski A spec could be commented out for now but still usable as a reference for a developer. :)